tests/test_flask_logging.py

```diff
@@ -74,11 +74,18 @@ def test_logging_without_request():
     logger.info('works')
     assert check_log_record(stream, JOB_LOG_SCHEMA, {'msg': v_str('works')}) == {}
 
+def test_custom_field_loggin():
+    """ Test custom fields are generated """
+    app = Flask(__name__)
+    _set_up_flask_logging(app)
+    logger, stream = config_logger('main.logger')
+    logger.info('works', extra={'cf1': 'yes'})
+    assert check_log_record(stream, CUST_FIELD_SCHEMA, {}) == {}
```

Comment:
Would it make sense to additionally check the contents of the JSON for correct `k`, `v` and `i`?

Reply:
Yes; however there seems to be a bug in the json validator that we use:

```python
from json_validator import JsonValidator
schema =  { '#cf': {
      'type': dict,
      'properties': {
          'string': {'type' : list, 'items': {
              'type': dict,
              'properties': {
                  'v': {'format': r'.*'},
                  'k': {'format': r'.*'},
                  'i': {'type': int, 'gt': -1},
              }
          }}
      }
    }}

json1 ={"#cf": {"string": [{"k": "key", "v": "value", "i": 0}]}}
JsonValidator(schema).validate(json1)
# ({'#cf': {'string': [{'v': 'value', 'k': 'key'}]}}, {})


json2 = {"#cf": {"string": [{"k": "key", "v": "value", "i": 1}]}}
JsonValidator(schema).validate(json2)
# ({'#cf': {'string': [{'v': 'value', 'k': 'key', 'i': 1}]}}, {})

```

Whenever i is zero it is not outputted and this messes up the validation. 
After this is fixed in sonic182-json-validator we can add the values check as well.